server/task/task-distributed/src/main/scala/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionDAO.scala

```diff
@@ -55,20 +57,43 @@ class CassandraTaskExecutionDetailsProjectionDAO @Inject()(session: Session, typ
 
   private val listStatement = session.prepare(select().from(TABLE_NAME))
 
-  def saveDetails(details: TaskExecutionDetails): Mono[Void] = cassandraAsyncExecutor.executeVoid(
-    insertStatement.bind
+  def saveDetails(details: TaskExecutionDetails): Mono[Void] = {
+    val boundStatement =  insertStatement.bind
       .setUUID(TASK_ID, details.getTaskId.getValue)
       .setString(TYPE, details.getType.asString())
       .setString(STATUS, details.getStatus.getValue)
       .setUDTValue(SUBMITTED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getSubmittedDate))
       .setString(SUBMITTED_NODE, details.getSubmittedNode.asString)
-      .setUDTValue(STARTED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getStartedDate).orElse(null))
-      .setString(RAN_NODE, details.getRanNode.map[String](_.asString).orElse(null))
-      .setUDTValue(COMPLETED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCompletedDate).orElse(null))
-      .setUDTValue(CANCELED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCanceledDate).orElse(null))
-      .setString(CANCEL_REQUESTED_NODE, details.getCancelRequestedNode.map[String](_.asString).orElse(null))
-      .setUDTValue(FAILED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getStartedDate).orElse(null))
-      .setString(ADDITIONAL_INFORMATION, serializeAdditionalInformation(details).orElse(null)))
+
+    val optionalFieldsToBind = List((STARTED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getStartedDate)),
+      (RAN_NODE, details.getRanNode.map[String](_.asString)),
+      (COMPLETED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCompletedDate)),
+      (CANCELED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCanceledDate)),
+      (CANCEL_REQUESTED_NODE, details.getCancelRequestedNode.map[String](_.asString)),
+      (FAILED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getFailedDate)),
+      (ADDITIONAL_INFORMATION, serializeAdditionalInformation(details))
+    )
+
+    val fullyBoundStatement = optionalFieldsToBind.foldLeft(boundStatement)((statement, field) => {
+      val (fieldName, fieldValue) = field
+      bindOptionalValue(statement, fieldName, fieldValue)
+    })
+
+    cassandraAsyncExecutor.executeVoid(fullyBoundStatement);
+  }
+
+  private def bindOptionalValue(statement: BoundStatement, fieldName: String, fieldValue: Optional[_]) = {
```

Comment:
I would rather define the method to use in the tuple to avoid the type discovery (we know the type at compile-time here)